enterprise/server/scheduling/executor_handle/executor_handle.go

```diff
@@ -196,6 +198,13 @@ func (h *registrationAndTasksExecutorHandle) handleTaskReservationResponse(respo
 }
 
 func (h *registrationAndTasksExecutorHandle) EnqueueTaskReservation(ctx context.Context, req *scpb.EnqueueTaskReservationRequest) (*scpb.EnqueueTaskReservationResponse, error) {
+	req, ok := proto.Clone(req).(*scpb.EnqueueTaskReservationRequest)
```

Comment:
maybe add a comment about why we clone the proto here?

Also I assume it's not an issue but would a caller ever have set this field before calling EnqueueTaskReservation?

Reply:
Done. I don't think there's a reason for the caller to have set this field prior to this function.